docs/guides/server/configuration.adoc

```diff
@@ -8,56 +8,53 @@ summary="Understand how to configure and start Keycloak">
+When an option is set in more than one source, the one that comes first in the list determines the value for that option. For example, the value for an option set by a command-line parameter has a higher priority than an environment variable for the same option.
 
-=== Example: Configuring the db-url-host parameter.
+=== Example: Configuring the db-url-host parameter
+
+The following example shows how the `db-url` value is set in three configuration sources:
 
 |===
 |*Source* | *Format*
 
-|CLI
-|--db-url=cliValue
+|Command line parameters
+|`--db-url=cliValue`
 
-|Environment Variable
-|KC_DB_URL=envVarValue
+|Environment variable
+|`KC_DB_URL=envVarValue`
 
 |Configuration file
-|db-url=confFileValue
+|`db-url=confFileValue`
 |===
 
-In the example above, the `db-url` value is set in all three configuration sources. The actual value that is used at startup would be the `cliValue`. If `--db-url=cliValue` is not used, the used value would be `KC_DB_URL=envVarValue`, and last but not least the `db-url=confFileValue` would be used when no environment variable with the same Key is present. When this value is specified in a user defined configuration file and in `conf/keycloak.conf`, the value from the user defined configuration file takes precedence.
+Based on the priority of application, the value that is used at startup is `cliValue`, because the command line is the highest priority.
 
-== Configuration Format
-The configuration follows a "unified-per-source" format, that is easily translatable from one configuration source to another:
+If `--db-url=cliValue` had not been used, the applied value would be `KC_DB_URL=envVarValue`. If the value were not applied by either the command line or an environment variable, `db-url=confFileValue` would be used. However, if this value had been set in a user defined configuration file, that value would take precedence over the `conf/keycloak.conf` file.
 
-.Command-line parameter format
-Values for the command-line are following the `--<key-with-dashes>=<value>` format. For some values, there's also a `-<abbreviation>=value` shorthand.
+== Formats for configuration
+The configuration uses a _unified-per-source_ format, which simplifies translation of a key/value pair from one configuration source to another.
 
-.Environment variable format
-Values for environment variables are following the uppercased `KC_<key_with_underscores>=<value>` format.
+Command-line parameter format:: Values for the command-line use the `--_<key-with-dashes>_=_<value>_` format. For some values, an `-_<abbreviation>_=_<value>_` shorthand also exists.
 
-.Configuration file format
-Values that go into the configuration file are following the `<key-with-dashes>=<value>` format.
+Environment variable format:: Values for environment variables use the uppercased `KC_<key_with_underscores>=<value>` format.
```

Comment:
Thanks, @vmuzikar. Your comment is correct, but generating the output interprets the first underscore as an actual underscore, not a formatting element to create an italics output.  I have not found a workaround.

Reply:
The first underscore can be marked as pass-through by quoting it with `++`. The `<key_with_underscores>` then needs to be wrapped in double `__` to avoid confusion with the single underscores it contains. For `<value>` single underscores are sufficient. 

This leads to the following snippet: 

```
Environment variable format:: Values for environment variables use the uppercased `KC++_++__<key_with_underscores>__=_<value>_` format.
```

which renders as

![image](https://user-images.githubusercontent.com/3957921/229291880-ba1cae46-9b63-4e06-8685-46c7d09be54b.png)